packages/php-wasm/logger/src/lib/log-collector.ts

```diff
@@ -0,0 +1,185 @@
+/// <reference lib="WebWorker" />
+
+import {
+	PHPRequestErrorEvent,
+	UniversalPHP,
+} from '@php-wasm/universal/src/lib/universal-php';
+import type { Logger } from './logger';
+
+let windowConnected = false;
+
+/**
+ * Log Windows errors.
+ *
+ * @param loggerInstance The logger instance
+ * @param ErrorEvent event
+ */
+function logWindowError(loggerInstance: Logger, event: ErrorEvent) {
+	loggerInstance.addLogEntry(
+		`${event.message} in ${event.filename} on line ${event.lineno}:${event.colno}`,
+		'Error'
+	);
+}
+
+/**
+ * Log unhandled promise rejections.
+ *
+ * @param loggerInstance The logger instance
+ * @param PromiseRejectionEvent event
+ */
+function logUnhandledRejection(
+	loggerInstance: Logger,
+	event: PromiseRejectionEvent
+) {
+	// No reason was provided, so we can't log anything.
+	if (!event?.reason) {
+		return;
+	}
+	const message = event?.reason.stack ?? event.reason;
+	loggerInstance.addLogEntry(message, 'Error');
+}
+
+/**
+ * Register a listener for service worker messages and log the data.
+ * The service worker will send the number of open Playground tabs.
+ *
+ * @param loggerInstance The logger instance
+ */
+function addServiceWorkerMessageListener(loggerInstance: Logger) {
+	navigator.serviceWorker.addEventListener('message', (event) => {
+		if (event.data?.numberOfOpenPlaygroundTabs !== undefined) {
+			loggerInstance.addContext({
+				numberOfOpenPlaygroundTabs:
+					event.data.numberOfOpenPlaygroundTabs,
+			});
+		}
+	});
+}
+
+/**
+ * Collect errors from JavaScript window events like error and log them.
+ * @param loggerInstance The logger instance
+ */
+export function collectWindowErrors(loggerInstance: Logger) {
+	// Ensure that the window events are connected only once.
+	if (windowConnected) {
+		return;
+	}
+	addServiceWorkerMessageListener(loggerInstance);
+	if (typeof window === 'undefined') {
+		return;
+	}
+
+	window.addEventListener('error', (event) =>
+		logWindowError(loggerInstance, event as ErrorEvent)
+	);
+	window.addEventListener('unhandledrejection', (event) =>
+		logUnhandledRejection(loggerInstance, event as PromiseRejectionEvent)
+	);
+	window.addEventListener('rejectionhandled', (event) =>
+		logUnhandledRejection(loggerInstance, event as PromiseRejectionEvent)
```

Comment:
The data flow like `rejectionhandled` -> `logUnhandledRejection` tells me there's something off with naming here.

Reply:
I was just lazy and reused the function. The name is now updated to `logPromiseRejection`.